clang/cmake/modules/AddClang.cmake

```diff
@@ -108,6 +108,10 @@ macro(add_clang_library name)
   endif()
   llvm_add_library(${name} ${LIBTYPE} ${ARG_UNPARSED_ARGUMENTS} ${srcs})
 
+  if(NOT ARG_SHARED AND NOT ARG_STATIC)
+    target_compile_definitions("obj.${name}" PRIVATE CLANG_EXPORTS)
```

Comment:
Why is this being added?

Reply:
This is the way clang component libraries are indirectly defined, this commit sort of related https://github.com/fsfod/llvm-project/commit/f539bdcef3d299e7ea658c4575b860a082bcda71 and this flang commit sort of explains it as well https://github.com/llvm/llvm-project/commit/424ad163c715e478d37cfd1efaae148cdd6aa815